test/draft-published.js

```diff
@@ -115,6 +115,20 @@ describe('Draft / Published', function() {
               }
             }
           }
+        },
+        '@apostrophecms/doc-type': {
+          extendMethods(self) {
+            return {
+              emit(_super, message, req, data, ...args) {
+                // Sort of a way to spy `emit` function
+                self.emitStack = {
+                  ...(self.emitStack || {}),
+                  [message]: data
+                };
```

Comment:
Are we ok to do this kind of things?
We should probably use [sinon](https://sinonjs.org/releases/latest/spies/) to do this, wdyt?

Reply:
Why not just write a handler for `beforeUnpublish` instead?

Reply:
(Sinon is nice, but it seems like writing a beforeUnpublish handler would be a fine way to verify the event was emitted.)

Reply:
Yes that makes actually way more sense 😄